### ISSUE TITLE
Advice from a "bad" assembly

### ISSUE THREAD

**Issue by frsepulveda** [marbl/canu#2221]
Hi everyone! We need some advice from our PacBio assembly. We have PacBio data from a red algae with a haploid genome of 100Mb. We expect to have around 300 scaffold (results from other similar and related red algae). But our assembly with canu result about 3000 scaffolds! We already try purge_dups after the assembly but there is not a lot of difference. Further, we try use minReadLength=3500 minOverlapLength=3500 (idea from a flye assembly). Also, we cleaned our raw reads with centrifuge before the assembly.  What can i do to obtain less Scaffolds for my assembly?

Here the are some statics: 

- Raw statics of subreads (before Canu): 
sample name: 2A1
Total bases: 62,51Gb
Total reads: 5097442
Average leng: 12263
MaxLen: 328942
N50Len: 12836

coverage: 800X. 

- Statics from Canu Assembly:  

canu -p mazzalam -d mazzalam-pacbio genomeSize=100m -pacbio-raw ../limpieza/1A1_CLEAN.fa useGrid=false maxInputCoverage=200 maxThreads=30 maxMemory=200

```
--    G=9779650021                       sum of         ||               length     num
--    NG         length     index       lengths           ||                range    seqs
--    ----- ------------ --------- ------------          ||  ------------------- -------
--    00010        15018     60502    977966239    ||       1000-2181        64196|----------------
--    00020        14024    128018   1955932841  ||       2182-3363        51172|-------------
--    00030        13183    199983   2933897689  ||       3364-4545        42185|-----------
--    00040        12676    275742   3911867453  ||       4546-5727        37848|----------
--    00050        12289    354124   4889825971  ||       5728-6909        38166|----------
--    00060        11947    434851   5867798471  ||       6910-8091        28761|-------
--    00070        11619    517859   6845755321  ||       8092-9273        22428|------
--    00080        11230    603365   7823723765  ||       9274-10455       19857|-----
--    00090         7644    702013   8801686737  ||      10456-11637      136550|---------------------------------
--    00100         1000    954226   9779650021  ||      11638-12819                                                                                                                                      262366|---------------------------------------------------------------
--    001.000x              954227   9779650021    ||      12820-14001      120836|------------------------------
--                                                                       ||      14002-15183       77657|-------------------
--                                                                       ||      15184-16365       38127|----------
--                                                                       ||      16366-17547        9984|---
--                                                                       ||      17548-18729        1661|-
--                                                                       ||      18730-19911         615|-

[UNITIGGING/OVERLAPS]
--   category            reads     %          read length        feature size or coverage  analysis
--   ----------------  -------  -------  ----------------------  ------------------------  --------------------
--   middle-missing        104    0.03    11400.32 +- 2714.66       1295.10 +- 1490.28    (bad trimming)
--   middle-hump          1628    0.53     8045.60 +- 4061.11        196.66 +- 688.76     (bad trimming)
--   no-5-prime              4021    1.30     9424.21 +- 4040.93        139.10 +- 513.39     (bad trimming)
--   no-3-prime              2320    0.75    10044.06 +- 3840.78        150.77 +- 594.05     (bad trimming)
--
--   low-coverage        24503    7.91     8145.85 +- 4099.55          5.71 +- 2.55       (easy to assemble, potential for lower quality consensus)
--   unique                  20282    6.55    11975.92 +- 3145.88         37.53 +- 12.34      (easy to assemble, perfect, yay)
--   repeat-cont          91102   29.43    12719.60 +- 1924.22        242.57 +- 446.99     (potential for consensus errors, no impact on assembly)
--   repeat-dove         14766    4.77    14612.45 +- 1062.19        176.37 +- 226.99     (hard to assemble, likely won't assemble correctly or even at all)
--
--   span-repeat           19492    6.30    12735.63 +- 2330.15       5376.62 +- 3936.49    (read spans a large repeat, usually easy to assemble)
--   uniq-repeat-cont    79462   25.67    12679.22 +- 1856.50                             (should be uniquely placed, low potential for consensus errors, no impact on assembly)
--   uniq-repeat-dove    24507    7.92    14148.32 +- 1282.86                             (will end contigs, potential to misassemble)
--   uniq-anchor         27191    8.78    13208.61 +- 1553.15       3528.56 +- 3203.63    (repeat read, with unique section, probable bad read)

[UNITIGGING/CONSENSUS]
-- Found, in version 2, after consensus generation:
--   contigs:      2185 sequences, total length 85787152 bp (including 683 repeats of total length 11288485 bp).
--   bubbles:      397 sequences, total length 6902736 bp.
--   unassembled:  30338 sequences, total length 258159982 bp.
--
-- Contig sizes based on genome size 100mbp:
--
--            NG (bp)  LG (contigs)    sum (bp)
--         ----------  ------------  ----------
--     10      265434            28    10102024
--     20      181072            74    20114845
--     30      135142           138    30059221
--     40       89619           229    40021761
--     50       58140           370    50043134
--     60       34336           596    60001601
--     70       20037           986    70002309
--     80       13504          1613    80001878
--
```

**Comment by frsepulveda** [comment]
[mazzalam.report.txt](https://github.com/marbl/canu/files/11172398/mazzalam.report.txt)

Thanks for the advice! 
Hi! here is attached the canu report from my run!

**Comment by skoren** [comment]
Those reads seem quite short for a CLR run, they mostly are 10-12kb which isn't great for assembly. It almost looks like a HiFi library. It also looks like there's only 100x coverage and 10gb of input not 65. Was the data pre-filtered somehow for length and or coverage?

**Comment by frsepulveda** [comment]
> Those reads seem quite short for a CLR run, they mostly are 10-12kb which isn't great for assembly. It almost looks like a HiFi library. It also looks like there's only 100x coverage and 10gb of input not 65. Was the data pre-filtered somehow for length and or coverage?

Not, our data weren't pre-filtered. There are the subreads of our sequencing. We only cleaned before the assembly. We also try to assembly a sample without cleaned it but we had similar results. 

Statics from our sequencing service:

![image](https://user-images.githubusercontent.com/128161915/230439680-eac15e5e-6e51-4105-9fd2-bc0ba38c19da.png)

![image](https://user-images.githubusercontent.com/128161915/230439927-532ca674-9c34-4ca2-8f20-8d20cb81bf55.png)
(this one was used to the canu assembly)

**Comment by skoren** [comment]
I think you should treat these as HiFi data based on these tables. If the polymerase read length mean is 100+kb and the subread length is 12kb you should get 8-10 passes for each read which would convert to HiFi. You'd likely end up with a much better assembly.

**Comment by frsepulveda** [comment]
> I think you should treat these as HiFi data based on these tables and run the ccs processing software. If the polymerase read length mean is 100+kb and the subread length is 12kb you should get 8-10 passes for each read which would convert to HiFi. You'd likely end up with a much better assembly.

Thanks skoren! I'll try and i keep you inform. A quick question: what is the ccs processing software?

**Comment by skoren** [comment]
It used to be called unanimity but not sure now, pacbio has some documentation here: https://ccs.how. You would input the subreads.bam which generates a new ccs.bam and you can extract fastq from that and run hicanu or hifiasm, yes. Your coverage will be much less than 800x (most likely 50-60x) but should still be enough for assembly.

**Comment by skoren** [comment]
Comparing N50 stats for different sized assemblies is meaningless since the denominator will be different, especially when one is 60mb and one is over 100mb. How confident are you that 100mb is the genome size? Is it possible that it's really 60mb and canu is giving you both haplotypes (about 120mb asm) while flye collapsed them and is outputting a single one? There are some suggestions on the FAQ about tuning parameters to smash or separate haplotypes, depending on what you want.

Your best bet given the really short read lengths you have is still to maximize their accuracy which is what the CCS conversion does. What happened when you tried to convert the reads to HiFi, what parameters/etc did you use? How many reads converted and what coverage did you end up with? Did you convert all the libraries you listed above or just one?

**Comment by frsepulveda** [comment]
> There are some suggestions on the FAQ about tuning parameters to smash or separate haplotypes, depending on what you want.
> 
> Your best bet given the really short read lengths you have is still to maximize their accuracy which is what the CCS conversion does. What happened when you tried to convert the reads to HiFi, what parameters/etc did you use? How many reads converted and what coverage did you end up with? Did you convert all the libraries you listed above or just one?

haha! yes, for a moment we think that maybe our genome size was wrong. Fortunately, we also have short reads from the same DNA samples and use k-mer genome size estimation and was around 98Mb (Haploid).

Regarding the CSS conversion, i dont have the information here (im not in the lab) but i could looking it and post here from advice.

**Comment by frsepulveda** [comment]
Hi!

I just re-runing from the assembly step wit corrected error rate higher (0.085) and have similar results (mazzalamerr085.report). The quast result show that we just has some few less scaffold (from 2582 to 2508). 
[mazzalamerr085.report.txt](https://github.com/marbl/canu/files/11304382/mazzalamerr085.report.txt). 

Since with this sample I have >100X coverage. Now i'm trying with other sample using: "correctedErrorRate=0.085 corMhapSensitivity=normal" following this FAQ advice:

"Based on an A. thaliana [dataset](http://www.pacb.com/blog/sequel-system-data-release-arabidopsis-dataset-genome-assembly/), and a few more recent mammalian genomes, slightly increase the maximum allowed difference from the default of 4.5% to 8.5% with correctedErrorRate=0.085 corMhapSensitivity=normal. Only add the second parameter (corMhapSensivity=normal) if you have >50x coverage."

canu -p mazzalam -d mazzalam-pacbio genomeSize=100m -pacbio-raw ../data/1A1.fa maxThreads=30 maxMemory=200 correctedErrorRate=0.085 corMhapSensitivity=normal
 

Also i'm re-running ccs software (but we are pretty sure that we dont have HiFI-PacBio data) because someone deleted my previous ccs running :( (I use a shared cluster).
